### PR TITLE
88 controller only

### PR DESCRIPTION
**Pull request by hinchliff**
Splitting apart the pull request in #97, this pull request is essentially only the modifications to the Controller.  The only real change to functionality in the controller was the addition of the ticketedRequestedEntities to createSlice() and modifySlice(), although some additional code was reorganized in OrcaXmlrpcHandler.

Other changes were mostly related to adding some unit tests for createSlice() and modifySlice().

### CODE REVIEW

**Comment by hinchliff** [comment]
Please submit your code review on GitHub prior to tomorrow's meeting.  #97 has some suggestions on the code review process.

File: controllers/xmlrpc/config/controller.properties [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+log4j.appender.file.File=logs/controller.log
+
 # stdout logging
 log4j.appender.stdout = org.apache.log4j.ConsoleAppender
 log4j.appender.stdout.layout = org.apache.log4j.PatternLayout
 log4j.appender.stdout.layout.ConversionPattern = %d [%t] %-5p %c - %m%n
+
 # controller log level
-log4j.logger.controller=DEBUG
-log4j.logger.orca=DEBUG
```
Comment (ibaldin) [inline controllers/xmlrpc/config/controller.properties]: I don't rememeber what this logger was for. It maybe relevant to actors only.

File: controllers/xmlrpc/build.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 <target name="package" description="Create .tar.gz for XMLRPC controller and daemon based on appassembler outputs" depends="clean">
 	<echo message="Building a .tgz file of XMLRPC application executable" />
-	<tar destfile="${xmlrpctar}" basedir="${basedir}/xmlrpc" includes="bin/* repo/**" compression="gzip" />
+	<tar destfile="${xmlrpctar}" basedir="${basedir}" includes="bin/* repo/**" compression="gzip" />
```
Comment (ibaldin) [inline controllers/xmlrpc/build.xml]: Will this affect the way RPMs are built?

File: controllers/xmlrpc/config/controller.properties [hunk 3]
```diff
@@ -41,19 +43,26 @@ ndl.converter.url=http://geni.renci.org:12080/ndl-conversion/, http://bbn-hn.exo
 #xmlrpc.controller.defaultBroker=ndl-broker
 
 # you can specify a port on which the controller listens (default 9443)
-xmlrpc.controller.port=11443
+xmlrpc.controller.port=10443
```
Comment (ibaldin) [inline controllers/xmlrpc/config/controller.properties]: 11443 is our 'standard' port. What's the reason here?

File: redhat/orca-iaas.spec [hunk 4]
```diff
@@ -211,7 +211,7 @@ cp -R server/orca/config %{buildroot}%{conf_dir}/sm-14080
 cp -R server/orca/lib %{buildroot}%{conf_dir}/sm-14080
 mkdir -p %{buildroot}%{conf_dir}/sm-14080/ssl
 # Populate controller-11080
-cp -R controllers/xmlrpc/xmlrpc/config %{buildroot}%{conf_dir}/controller-11080
+cp -R controllers/xmlrpc/config %{buildroot}%{conf_dir}/controller-11080
```
Comment (ibaldin) [inline redhat/orca-iaas.spec]: I guess that answered my question above

File: controllers/xmlrpc/src/main/java/orca/controllers/xmlrpc/OrcaXmlrpcHandler.java [hunk 5]
```diff
@@ -1100,6 +1069,23 @@ public OrcaXmlrpcHandler() {
 	 * @param ndlSlice
 	 * @return
 	 */
+	protected Map<String, Object> getRequestedEntities(XmlrpcControllerSlice ndlSlice) {
```
Comment (ibaldin) [inline controllers/xmlrpc/src/main/java/orca/controllers/xmlrpc/OrcaXmlrpcHandler.java]: I think this should be a method on XmlrpcControllerSlice

File: controllers/xmlrpc/src/test/java/orca/controllers/mock/MockOrcaConnectionFactory.java [hunk 6]
```diff
@@ -48,24 +48,12 @@ public MockOrcaConnectionFactory(String url, String user, String password, ID id
          * @throws Exception
          */
     public synchronized IOrcaServiceManager getServiceManager() throws Exception {
-        // obtain a proxy to the container.
```
Comment (ibaldin) [inline controllers/xmlrpc/src/test/java/orca/controllers/mock/MockOrcaConnectionFactory.java]: These are probably left by Jeff when he was originally writing it. There's some historical value here :-)

**Comment by ibaldin** [comment]
I'm done with my comments here.

**Pull request merged**